ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+        This function assumes that flexy-ocs-private repo has been
+        already cloned
+
+        """
+        config_parser = configparser.ConfigParser()
+        # without this,config parser would convert all
+        # uppercase keys to lowercase
+        config_parser.optionxform = str
+        env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
+
+        with open(env_file, "r") as fp:
+            # we need to add section to make
+            # config parser happy, this will be removed
+            # while writing back to file
+            file_content = "[root]\n" + fp.read()
+
+        config_parser.read_string(file_content)
+        # Iterate over config_parser keys, if same key is present
+        # in user supplied dict update config_parser
+        for ele in config_parser.items('root'):
+            if ele[0] in config.ENV_DATA:
+                # For LAUNCHER_VARS we need to merge the
+                # user provided dict with default obtained
+                # from env file
+                if ele[0] == 'LAUNCHER_VARS':
+                    merge_dict(config.ENV_DATA[ele[0]], ele[1])
+                config_parser['root'][ele[0]] = config.ENV_DATA[ele[0]]
+                logger.info(f"env updated {ele[0]}:{config.ENV_DATA[ele[0]]}")
+
+        # write the updated config_parser content back to the env file
+        tmp_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, f"{constants.FLEXY_DEFAULT_ENV_FILE}.tmp"
+        )
+        with open(tmp_file, "w") as fp:
+            f = io.StringIO()
+            config_parser.write(f)
+            f.seek(0)
+            # eliminate first line which has section [root]
+            # last line ll have a '\n' so that needs to be
+            # removed
+            fp.write("".join(f.readlines()[1:-1]))
+
+        # Move this tempfile to original file
+        os.rename(
+            tmp_file, env_file
+        )
+
+
+class FlexyBaremetalPSI(FlexyBase):
+    """
+    A specific implementation of Baremetal with PSI using flexy
+    """
+    def __init__(self):
+        super().__init__()
+        if not config.ENV_DATA.get('template_file_path'):
+            self.template_file = os.path.join(
+                constants.OPENSHIFT_MISC_BASE,
+                f"aos-{get_ocp_version('_')}",
+                constants.FLEXY_BAREMETAL_UPI_TEMPLATE
+            )
+        else:
+            self.template_file = config.ENV_DATA.get('template_file_path')
+
+    def deploy_prereq(self):
+        """
+        Common flexy prerequisites like cloning the private-conf
+        repo locally and updating the contents with user supplied
+        values
+
+        """
+        if not os.path.exists(self.flexy_mnt_host_dir):
+            os.mkdir(self.flexy_mnt_host_dir)
+            os.chmod(self.flexy_mnt_host_dir, mode=0o777)
+        self.clone_and_unlock_ocs_private_conf()
```

Comment:
@dahorak @petr-balogh ,
By default we are referring to template file which lives inside openshift-misc repo, so here you are saying that either user has already checkout openshift-misc repo and modified template file ?

Reply:
@shylesh we've discussed this yesterday.  This comment refer to line above I guess for ` self.clone_and_unlock_ocs_private_conf()` , and in case the user will provide the absolute path to `flexy env` file  then no need to clone the repository locally for private repo.

Reply:
Addressed in the above comments